.github/workflows/benchmarks.yml

```
@@ -0,0 +1,66 @@
+# Run performance benchmarks
+#
+# Continuous benchmarking using pytest-codspeed. Measures the execution speed
+# of tests marked with @pytest.mark.benchmark decorator.
+
+name: Benchmarks
+
+on:
+  # Run on pushes to the main branch
+  push:
+    branches: [ main ]
+  # Uncomment the 'pull_request' line below to trigger the workflow in PR
+  pull_request:
```

Comment out this line before merging:
```suggestion
  # pull_request:
```

---

Actually, how about we set [`paths:`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) so that the benchmark runs when any code in clib/datasets/helpers/src is modified:
```suggestion
  pull_request:
    paths:
      - 'pygmt/clib/**'
      - 'pygmt/datasets/**'
      - 'pygmt/helpers/**'
      - 'pygmt/src/**'
```

---

Why not just use `- 'pygmt/**/*.py'` to include all files?

---

That won't work right if we want to include other files, e.g. `environment.yml` when bumping to new versions of GMT for example.

---

Just re-read https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-paths, specifically this part:

> The order that you define paths patterns matters:
>
>  - A matching negative pattern (prefixed with !) after a positive match will exclude the path.
>  - A matching positive pattern after a negative match will include the path again.

So it looks like we can add more paths after `!pygmt/tests/**`.

---

Ok, made a commit in dedfa7af7250762f473a81928a609b04198d11b1. Have applied these `paths:` restrictions to both the `push` and `pull_request` triggers, see if that looks ok.